src/core/init/initialize_media_source.ts

```diff
@@ -225,52 +230,74 @@ export default function InitializeOnMediaSource(
    * Wait for the MediaKeys to have been created before opening the MediaSource,
    * after that second step is done, ask the EMEManager to attach the MediaKeys.
    * Steps are done in that specific order to avoid compatibility issues.
+   *
+   * This Observable will emit when ready both the MediaSource and useful
+   * DRM-specific information.
    */
   const prepareMediaSource$ = emeManager$.pipe(
-    mergeMap((evt) => {
+    mergeScan((
+      acc : {
+        /** set to true once EME APIs have been initialized. */
+        isEmeReady : boolean;
+        /**
+         * ID identifying the current MediaKeys' system ID. Can be used to only
+         * send initialization data linked to that ID as an optimization measure.
+         */
+        drmSystemId? : string;
+      },
+      evt : IEMEManagerEvent | IEMEDisabledEvent
+    ) => {
       switch (evt.type) {
         case "eme-disabled":
         case "attached-media-keys":
-          return observableOf(undefined);
+          return observableOf({ isEmeReady: true,
+                                drmSystemId: acc.drmSystemId });
         case "created-media-keys":
-          return openMediaSource$.pipe(mergeMap(() => {
-            // Now that the MediaSource has been opened and linked to the media
-            // element we can attach the MediaKeys instance to the latter.
-            evt.value.attachMediaKeys$.next();
+          const drmSystemId = evt.value.initializationDataSystemId;
+          return openMediaSource$.pipe(
+            mergeMap(() => {
+              // Now that the MediaSource has been opened and linked to the media
+              // element we can attach the MediaKeys instance to the latter.
+              evt.value.attachMediaKeys$.next();
 
-            // If the `disableMediaKeysAttachmentLock` option has been set to
-            // `true`, we should not wait until the MediaKeys instance has been
-            // attached to start loading the content.
-            const shouldDisableLock = evt.value.options
-              .disableMediaKeysAttachmentLock === true;
-            return shouldDisableLock ? observableOf(undefined) :
-                                       EMPTY;
-          }));
+              // If the `disableMediaKeysAttachmentLock` option has been set to
+              // `true`, we should not wait until the MediaKeys instance has been
+              // attached to start loading the content.
+              const shouldDisableLock = evt.value.options
+                .disableMediaKeysAttachmentLock === true;
+              return shouldDisableLock ? observableOf({ isEmeReady: true,
+                                                        drmSystemId }) :
+                                         EMPTY;
+            }),
+            startWith({ isEmeReady: false, drmSystemId }));
```

Comment:
We could have done something that just return the drmSystemId, that would mean that the EME is ready, no ?

Reply:
No because it can also be `undefined`

Reply:
👍🏻